packages/edit-site/src/components/header/index.js

```diff
@@ -79,64 +80,92 @@ export default function Header( {
 	const displayBlockToolbar =
 		! isLargeViewport || deviceType !== 'Desktop' || hasFixedToolbar;
 
+	// Testing logic for hovered name...
+	const { hoveredBlockIds, getBlock, getBlockName } = useSelect(
+		( select ) => {
+			const { getHoveredBlocks, getBlock, getBlockName } = select(
+				'core/block-editor'
+			);
+
+			return {
+				hoveredBlockIds: getHoveredBlocks(),
+				getBlock,
+				getBlockName,
+			};
+		}
+	);
+	const nameToDisplay = useMemo( () => {
+		// Go through hovered blocks and see if one is of interest.
+		const templatePartBlockId = hoveredBlockIds.find(
```

Comment:
I wonder if we should make this a selector like `isBlockOfTypeHovered` or something. Maybe premature optimization though

It also seems like it might be expensive to run this filter whenever the hovered block state changes. I wonder if that could be optimized at all 🤔

Reply:
> It also seems like it might be expensive to run this filter whenever the hovered block state changes. I wonder if that could be optimized at all 🤔

I agree, that was my initial concern with this approach.  Alternatively, using a different store to just track the name we should display and only adding the event listeners to the blocks we care about would run a lot less often.